ray-operator/controllers/ray/common/service.go

```diff
@@ -40,6 +40,42 @@ func BuildServiceForHeadPod(cluster rayiov1alpha1.RayCluster, labels map[string]
 		annotations = make(map[string]string)
 	}
 
+	defaultAppProtocol := DefaultServiceAppProtocol
+	ports := getServicePorts(cluster)
+	if cluster.Spec.HeadGroupSpec.HeadService != nil {
+		// Use the provided HeadService
+		// Priority: HeadService > HeadGroupSpec.PodTemplateSpec > default labels
+		for k, v := range labels {
```

Comment:
You can take a look at #874 for more context about label priority.

Reply:
Thanks for the pointer -- can you help summarize the conclusion from that discussion as it relates to the current PR? I don't have much expertise here so I'd opt to just follow your recommendation.

Is the conclusion the following?
- Fill the `Selector` field with labels, according to the priority `HeadGroupSpec.PodTemplateSpec > default labels > HeadService`
- Fill the `Labels` field with `HeadService` labels only

Reply:
For selector, we should always honor `HeadGroupSpec.PodTemplateSpec`. If a mismatch exists between head Pod's label (`HeadGroupSpec.PodTemplateSpec`) and head service's selector, workers cannot talk to the head Pod anymore.

* Head Service:
  * selector: `default labels` (if overlaps exist, respect `HeadGroupSpec.PodTemplateSpec` and ignore `HeadService`.) 
    * We will only have 5 key-value pairs (the number of default labels) here.
  * labels: selector + `HeadService` (if overlaps exist, respect selector and ignore `HeadService`)
    * We may have more than 5 key-value pairs here. 

This is a bit different from the section ["My current thought"](https://github.com/ray-project/kuberay/pull/874#pullrequestreview-1264771952) because some users may use labels to find the head service.